cuda_core/cuda/core/experimental/_device.py

```diff
@@ -968,26 +969,34 @@ def __new__(cls, device_id=None):
 
         # important: creating a Device instance does not initialize the GPU!
         if device_id is None:
-            device_id = handle_return(runtime.cudaGetDevice())
-            assert_type(device_id, int)
-        else:
-            total = handle_return(runtime.cudaGetDeviceCount())
-            assert_type(device_id, int)
-            if not (0 <= device_id < total):
-                raise ValueError(f"device_id must be within [0, {total}), got {device_id}")
+            err, dev = driver.cuCtxGetDevice()
+            if err == 0:
```

Comment:
+1 would be good to use `driver.CUResult.CUDA_SUCCESS`?

Reply:
done in commit 708fd70f